### PR TITLE
Marketplace-ui

### PR DESCRIPTION
**Pull request by bbemis017**

### CODE REVIEW

**Comment by bbemis017** [comment]
Is it worth refactoring the card-grid into smaller components? The listing data is somewhat different and kinda makes the card grid hard to use. Or is there a better way to handle that?

**Comment by harrischristiansen** [comment]
The way I have typically seen this done is to add a "style", "context", or otherwise attribute to your smaller (card) component, and allow that component to vary it's design to fit that context. So if the context "battles" is passed in, the card will render with information more specific to it's battle properties, while if the context "marketplace" was passed in, it will render the card with the marketplace info.

We should also rename the component GridCard, and pass in just the card object, such that everything references directly to card.

File: frontend/src/actions/listings.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import apiFetch from './index';
+
+export const REQUEST_ALL_LISTINGS = 'REQUEST_ALL_lISTINGS';
```
Comment (harrischristiansen) [inline frontend/src/actions/listings.js]: `'REQUEST_ALL_lISTINGS'` -> `'REQUEST_ALL_LISTINGS'`

**Comment by nickysemenza** [comment]
I agree with what @harrischristiansen  said, but style and context are both react-ish keywords so something like 'type'/something else might be a more apt term for the prop.

**Review by harrischristiansen — CHANGES_REQUESTED** [review]

File: frontend/src/pages/Marketplace.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import React, { Component } from 'react';
+import CardGrid from '../components/CardGrid';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { fetchAllListings } from '../actions/listings';
+import CardFilterSort from '../components/CardFilterSort';
```
Comment (harrischristiansen) [inline frontend/src/pages/Marketplace.js]: This is imported but never used, throwing an error:

```
./src/pages/Marketplace.js
  Line 6:  'CardFilterSort' is defined but never used  no-unused-vars
```

**Review by harrischristiansen — CHANGES_REQUESTED** [review]

File: frontend/src/components/EtherPrice.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import React, { Component } from 'react';
```
Comment (nickysemenza) [inline frontend/src/components/EtherPrice.js:1]: (I'm guilty of not doing this on other components but...) For highly reusable components, PropTypes are handy for IDE code completion + sanity checking: https://reactjs.org/docs/typechecking-with-proptypes.html 

e.g.:`yarn add prop-types`

`import PropTypes from 'prop-types';`

```
EtherPrice. propTypes = {
    price: PropTypes.number.isRequired
}
```

Comment (nickysemenza) [reply frontend/src/components/EtherPrice.js:1]: also unrelated and optional, but FYI in react there are things called stateless functional components that can make components that don't use state a little more readable: 
[here's a good primer](https://hackernoon.com/react-stateless-functional-components-nine-wins-you-might-have-overlooked-997b0d933dbc)

File: frontend/src/components/Card.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import React, { Component } from 'react';
```
Comment (nickysemenza) [inline frontend/src/components/Card.js:1]: this looks 🔥overall; this component could also use some prop-types 😄 
You can even do fancy stuff like ` type: PropTypes.oneOf(['marketplace', 'collection']),

also, instead of just having strings for 'marketplace' and 'collection', a good paradigm would be do have this component also export a variable with the types, just like how the redux actions do.

E.g. 
```
export const CARD_TYPE_MARKETPLACE = 'marketplace';
export const CARD_TYPE_COLLECTION = 'collection';
```

and then the places that use and then you can do `<CardGrid type={CARD_TYPE_COLLECTION} />`

tl;dr minimizing 'magic' numbers/strings is good!

**Review by nickysemenza — CHANGES_REQUESTED** [review]
also looks like somehow your `node_modules` ended up a level up from frontend somehow? and got accidentally committed?

File: frontend/src/styles/App.scss [hunk 5]
```diff
@@ -23,12 +23,20 @@ body {
...
   top: 0;
-  background: rgba(0, 0, 0, 0.5);
   transition: 0.5s ease;
-  opacity: 0;
   font-size: 20px;
-  padding: 20px;
+}
+
+.overlay-background {
+  background: rgba(0, 0, 0, 0.5);
```
Comment (harrischristiansen) [inline frontend/src/styles/App.scss]: Define a fallback background above. Eg:
```
background: #000000; /* This is the fallback color, for browsers that don't support rgba */
background: rgba(0, 0, 0, 0.5); /* This will overwrite and be used as the background in supporting browsers */
```

File: frontend/src/styles/App.scss [hunk 6]
```diff
@@ -41,3 +49,14 @@ body {
 .details:hover {
   color: #0000ff;
 }
+
+.ether-price {
+
+  p {
+    display: inline-block;
```
Comment (harrischristiansen) [inline frontend/src/styles/App.scss]: Lets not overwrite all `<p>` elements to be inline-block. I'm not sure why we would want to do this, as this goes against the nature of `<p>` elements.

If there is some specific element you want to be an inline block, you can add the bootstrap inline-block class to it.

**Review by harrischristiansen — CHANGES_REQUESTED** [review]

File: frontend/src/components/EtherPrice.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+import PropTypes from 'prop-types';
+
+class EtherPrice extends Component {
+  convertToEth(value) {
+    return value / Math.pow(10, 13);
+  }
+  render() {
+    return (
+      <div className="ether-price">
+        <span className="">{this.convertToEth(this.props.price)} Eth</span>
```
Comment (nickysemenza) [inline frontend/src/components/EtherPrice.js]: empty quotes

**Review by nickysemenza — DISMISSED** [review]
overall i think this seems good to go? (except for those conflicts ofc)

**Review by harrischristiansen — APPROVED** [review]
